bot/cogs/reminders.py

```diff
@@ -421,6 +425,24 @@ async def delete_reminder(self, ctx: Context, id_: int) -> None:
             delivery_dt=None,
         )
 
+    async def _can_modify(self, ctx: Context, reminder_id: t.Union[str, int]) -> bool:
+        """
+        Check whether the reminder can be modified by the ctx author.
+
+        The check passes when the user is an admin, or if they created the reminder.
+        """
+        if with_role_check(ctx, Roles.admins):
+            return True
+
+        api_response = await self.bot.api_client.get(f"bot/reminders/{reminder_id}")
```

Comment:
Oh I didn't see the site PR. Well, that API change wasn't really necessary cause you could have passed it as a param.

Still, you should catch the 404.

Reply:
The error handler has custom api 404 handling which should be sufficient here https://github.com/python-discord/bot/blob/a2c05febd122db2d8f4e7d4743a08aef7c2fe5e0/bot/cogs/error_handler.py#L234-L239
The api didn't allow ids for filtering, so since it'd have to be changed anyway I decided to do it directly instead of a filter since ids are unique.

Reply:
It does allow IDs for filtering. Anyway, good point on the 404. This is good to go then.